dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -649,8 +649,9 @@ void
 SolverSIM::reinit() {
   int counter = 0;
   modeChangeType_t modeChangeType = model_->getModeChangeType();
+  if (modeChangeType == NO_MODE) return;
 
-  if (modeChangeType == DIFFERENTIAL_MODE)
+  if (modeChangeType < minimumModeChangeTypeForAlgebraicRestoration_)
```

Comment:
In theory, when you go through reinit() you already crossed a mode change otherwise you shouldn't go through it but well we can have a double check here.
But then we don't really need to have twice comparisons, modeChangeType < minimumModeChangeTypeForAlgebraicRestoration should make the job in any case.

Reply:
Solved